packages/eslint-plugin-react-hooks/src/RulesOfHooks.js

```diff
@@ -520,7 +530,8 @@ export default {
               if (
                 !cycled &&
                 pathsFromStartToEnd !== allPathsFromStartToEnd &&
-                !isUseIdentifier(hook) // `use(...)` can be called conditionally.
+                !isUseIdentifier(hook) && // `use(...)` can be called conditionally.
+                !isInsideDoWhileLoop(hook) // wrapping do/while loops are checked separately.
```

Comment:
Specifically excluding hooks inside `do/while` calls because we're handling those cases separately in the above  check.